framework/mesh/mesh_continuum/mesh_continuum.cc

```diff
@@ -931,103 +931,69 @@ bool
 MeshContinuum::CheckPointInsideCell(const Cell& cell, const Vector3& point) const
 {
   const auto& grid_ref = *this;
-  typedef Vector3 Vec3;
-  auto InsideTet = [](const Vec3& point, const Vec3& v0, const Vec3& v1, const Vec3& v2)
+
+  auto InsideTet = [](const Vector3& point, const Vector3& v0, const Vector3& v1, const Vector3& v2)
   {
     const auto& v01 = v1 - v0;
     const auto& v02 = v2 - v0;
-
     const auto n = v01.Cross(v02).Normalized();
     const auto c = (v0 + v1 + v2) / 3.0;
-
     const auto pc = point - c;
 
     if (pc.Dot(n) > 0.0)
       return true;
-    else
-      return false;
+
+    return false;
   };
 
-  bool inside = true;
   if (cell.Type() == CellType::SLAB)
   {
     const auto& v0 = grid_ref.vertices[cell.vertex_ids_[0]];
     const auto& v1 = grid_ref.vertices[cell.vertex_ids_[1]];
 
-    const auto v01 = v1 - v0;
-    const auto v0p = point - v0;
-
-    const double v0p_dot_v01 = v0p.Dot(v01);
-
-    if (not(v0p_dot_v01 >= 0 and v0p_dot_v01 < v01.Norm()))
-      inside = false;
-  } // slab
-
+    if (((v0.z - point.z) * (v1.z - point.z)) > 0.0)
+      return false;
+  }
   else if (cell.Type() == CellType::POLYGON)
```

Comment:
A comment on the logic in this check would be helpful as that it is not immediately obvious.

Reply:
Fixed.